### PR TITLE
[LS] Add FVM stdlib to Flow Integration

### LINKED ISSUE #331: [LS] Add InternalEVM stdlib Contract
### Issue to be solved

Because the `InternalEVM` standard library value is not a part of the Cadence runtime, but rather, part of the FVM, it is not known to the language server.  It is, however, valid to the FVM & developers wishing to get code-analysis while using the `EVM`/`InternalEVM` contract will be unable to.

This is a more general problem than just the `InternalEVM` contract, however, it seems to me that the `InternalEVM` contract is the only one that needs to be considered for the time being.

(related https://github.com/onflow/vscode-cadence/issues/574)

### Suggested Solution

Add an additional option to the server, `WithStandardLibraryValues`, or similar, that would accept additional values that may be declared as a part of the stdlib.

This could be used as a part of the LS flow integration, where the FVM standard library values are specified.  This way the FVM-specific stdlib stays decoupled from the Cadence LS implementation itself.

### PR DESCRIPTION
**Pull request by jribbink**
Closes #331 

## Description

Adds option to LS to accept additional stdlib values.  This option is used by flow integration to provide FVM-native stdlib values (e.g. InternalEVM contract right now... others in future?)

______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence-lint/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Comment by jribbink** [comment]
I wasn't able to add the InternalEVM Contract to the WASM build unfortunately.  Various downstream dependencies from the InternalEVM value declaration in flow-go (related to go-ethereum) do not build in WASM due to `x/sys` dependency, so it wasn't possible to import the FVM stdlib to the WASM build as a result.

I think this probably means that certain refactoring needs to be done in flow-go in order to avoid these dependencies... but not really sure how easy this is.

File: languageserver/test/index.test.ts [hunk 1]
```diff
@@ -308,6 +308,20 @@ describe("diagnostics", () => {
...
   )
 
+  test("InternalEVM contract exists", async() =>
+      testCode(
+          `
+            access(all)
+            fun main() {
+                // Checks that the InternalEVM contract exists
+                // Also that it has the correct value (i.e. the run function exists)
+                log(InternalEVM.run)
```
Comment (turbolent) [inline languageserver/test/index.test.ts:318]: The current implementation defines a `StandardLibraryValue` in [`FVMStandardLibraryValues`](https://github.com/onflow/cadence-tools/pull/332/files#diff-e4908bc6e5534823f44d290f658cbbbb8503ce1126ea58fd611503bbfcb3865dR29) and then declares it in general, for all locations in [`WithExtendedStandardLibraryValues`](https://github.com/onflow/cadence-tools/pull/332/files#diff-9dc030944615b63c919527ae35f799e2c63ff5137f18ef2165d07d808fd2b81fR295-R296).

There are two problems with this:
1. `InternalEVM` is just a current implementation detail of `EVM`. "normal" users of Cadence should not have to know about it, nor use it – we do not guarantee that this implementation detail will stay like it is, we might change the implementation of the `EVM` contract.
2. In addition to defining the `EVM` value, also the `EVM` type needs to be defined.

Both problems can be solved by using FVM's `stdlib.SetupEnvironment` function.

**Review by turbolent — COMMENTED** [review]
Nice!

File: languageserver/test/index.test.ts (hunk 1 above)
Comment (jribbink) [reply languageserver/test/index.test.ts:318]: Ah gotcha, I didn't realize that `InternalEVM` was only exposed to the EVM contract.  I'll figure out how to update this 🙂

Comment (turbolent) [reply languageserver/test/index.test.ts:318]: I had another quick look into this, but the LS is not using the `runtime` package (`Runtime`/`Environment`), so using `stdlib.SetupEnvironment` isn't really possible at the moment.

We should eventually refactor the LS, but for now, defining `InternalEVM` everywhere to allow the `EVM` contract to be imported seems good enough.

**Review by turbolent — APPROVED** [review]

**Pull request merged**